app/models/claim.rb

```diff
@@ -18,6 +18,7 @@ class Claim < ApplicationRecord
   has_many :secondary_representatives, class_name: 'Representative',
                                        through: :claim_representatives, source: :representative
   has_many :uploaded_files, through: :claim_uploaded_files
+  has_many :pre_allocated_file_keys, as: :allocated_to, dependent: :destroy, inverse_of: :allocated_to
```

Comment:
Allows pre allocation data (for the pdf) to be stored against the claim.  So the front end can be given the URL in advance of the file being generated.